### PR TITLE
feat(autocad/rhino): CNX-162 CNX-163 adds color proxies on send and receive

### PR DESCRIPTION
**Pull request by clairekuang**
Adds color proxies on send and receive to objects and layer in rhino and autocad.
Key changes:
- `ColorProxy` now replaces the functionality of the Layer.color property
- `ColorManager` introduced in autocad and rhino
- improvements on color conversions in autocad<>autocad: `ColorMethod.ByAci` is now preserved compared to DUI2
- color inheritance (by object/by layer) should be almost perfect between autocad/rhino: just missing "color by material" option

Tested autocad<> autocad, rhino<>rhino, autocad<>rhino.

### CODE REVIEW

**Review by adamhathcock — DISMISSED** [review]
Needs someone else but +1 as I need this

File: Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadColorManager.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  {
+    var count = 0;
+    foreach (ColorProxy colorProxy in colorProxies)
+    {
+      onOperationProgressed?.Invoke("Converting colors", (double)++count / colorProxies.Count);
+      foreach (string objectId in colorProxy.objects)
+      {
+        AutocadColor convertedColor = ConvertColorProxyToColor(colorProxy);
+
+        if (!ObjectColorsIdMap.TryGetValue(objectId, out AutocadColor _))
```
Comment (oguzhankoral) [inline Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadColorManager.cs:119]: If we won't use get value of `TryGetValue` we could use `ContainsKey` too, but it is minor.

File: Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadLayerManager.cs [hunk 2]
```diff
@@ -23,33 +25,69 @@ public AutocadLayerManager(AutocadContext autocadContext)
...
+  /// <returns>The name of the existing or created layer</returns>
+  public string CreateLayerForReceive(
+    Collection[] layerPath,
+    string baseLayerPrefix,
+    Dictionary<string, AutocadColor> objectColorIdMap
+  )
   {
-    string layerName = _autocadContext.RemoveInvalidChars(layerCollection.name);
+    string[] namePath = layerPath.Select(c => c.name).ToArray();
+    string layerName = _autocadContext.RemoveInvalidChars(baseLayerPrefix + string.Join("-", namePath));
```
[the remark this answers is not in the document]
Comment (clairekuang) [reply Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadLayerManager.cs:64]: did find 2 other bugs though, which were fixed

**Review by oguzhankoral — APPROVED** [review]
Tested, acad <> acad, acad <>rhino and rhino <>rhino

**Review by JR-Morgan — APPROVED** [review]

**Pull request merged**